fawltydeps/settings.py

```diff
@@ -236,17 +237,17 @@ def create(cls, cmdline_args: argparse.Namespace) -> Settings:
         """
         args_dict = cmdline_args.__dict__
 
-        base_paths = set(getattr(cmdline_args, "basepaths", []))
-        if base_paths:
-            code_paths = args_dict.setdefault("code", base_paths)
-            deps_paths = args_dict.setdefault("deps", base_paths)
-            pyenv_paths = args_dict.setdefault("pyenvs", base_paths)
-            if base_paths not in (code_paths, deps_paths, pyenv_paths):
+        search_paths = set(getattr(cmdline_args, "search_paths", []))
+        if search_paths:
+            code_paths = args_dict.setdefault("code", search_paths)
+            deps_paths = args_dict.setdefault("deps", search_paths)
+            pyenv_paths = args_dict.setdefault("pyenvs", search_paths)
+            if search_paths not in (code_paths, deps_paths, pyenv_paths):
                 msg = (
-                    "All four path specifications (code, deps, pyenvs, and base)"
-                    f"have been used. Use at most 3. basepaths={base_paths}, "
-                    f"code_paths={code_paths}, deps_paths={deps_paths}, "
-                    f"pyenv_paths={pyenv_paths}"
+                    "All four path specifications (code, deps, pyenvs, and"
+                    "search_paths) have been used. Use at most 3."
+                    f"search_paths={search_paths}, code_paths={code_paths}, "
+                    f"deps_paths={deps_paths}, pyenv_paths={pyenv_paths}"
```

Comment:
Not to be addressed as part of this PR, but this limitation confuses me.
I think a warning would be better here, saying that `search_paths` will be ignored because code_paths, deps_paths and pyenv_paths are already populated.
Or just an expanded explanation about search_paths being used as a default for all these values.
Not quite sure what rn.

Reply:
Yeah, this code is a bit clumsy. I think I agree with you that a warning is better here. The fix is relatively quick, so I'll just add  it to this PR.

Reply:
Fixed in [220e802](https://github.com/tweag/FawltyDeps/pull/489/commits/220e8023e0e956a51f53ea559fcdb71ca597d84c).